libraries/SPIFFS/src/SPIFFS.cpp

```diff
@@ -43,23 +43,29 @@ bool SPIFFSImpl::exists(const char* path)
     return (f == true) && !f.isDirectory();
 }
 
-SPIFFSFS::SPIFFSFS() : FS(FSImplPtr(new SPIFFSImpl()))
+SPIFFSFS::SPIFFSFS() : FS(FSImplPtr(new SPIFFSImpl())), partitionLabel_(NULL)
 {
 
 }
 
+SPIFFSFS::~SPIFFSFS()
+{
+    delete partitionLabel_;
```

Comment:
add a not-null check for this before delete as it may throw an exception otherwise

Reply:
The standard says it is safe to call `delete` on `NULL`. Is this different on micro controllers?

Reply:
It's more of a general check and ensures that the underlying memory manager doesn't throw for a null being passed in (though you are right, it shouldn't!)

Reply:
Thanks for your hint :) Done!

Reply:
I replaced `delete` with `free` according to [https://stackoverflow.com/a/39935254/1433564](this StackOverflow answer). I hope this is correct.